components/supervisor/pkg/supervisor/supervisor.go

```diff
@@ -217,6 +217,7 @@ func Run(options ...RunOption) {
 	)
 	if cfg.DesktopIDE != nil {
 		desktopIdeReady = &ideReadyState{cond: sync.NewCond(&sync.Mutex{})}
+		portMgmt.AddInternalPort(uint32(cfg.DesktopIDE.ReadinessProbe.HTTPProbe.Port))
```

Comment:
I am not sure that using port is a proper way. It is not necessary defined and probe is not used like configuration to start an ide.

There should be something like `cfg.DesktopIDEPort`?

Reply:
@corneliusludmann could you point up from where 24000 port is coming from? It is hardcoded port used by IntelliJ products always? So probe indeed is the best way to read this info in code?

Reply:
ok, I just see that for other ports we configure them during workspace creation via https://github.com/gitpod-io/gitpod/blob/6d33af4106c40ae6ecb11bb61e0478a09929ea6f/components/ws-manager/pkg/manager/manager.go#L72-L73 and wonder whether we missing here `DesktopIDEPort` here. As far as I know the desktop IDE always need some kind of shell script serving index.html file and accepting the port to allow switching to desktop IDE, but not sure what was the motivation to pass it from outside.

@csweichel Should we add in ws-manager too?

Reply:
I'm not sure it's worth configuring these anymore. They're pretty much convention at this point, i.e. configuring the ports here would not work by itself.

Reply:
I meant if I try to access supervisor or web IDE endpoint on the port location like `23000-<workspace-id` it is no going to work.  It is strange that `24000` port behaves differently. I would assume it should be at least hardcoded in supervisor and ws-proxy, and supervisor should pass it desktop IDE startup code instead of hardcoding the same port in each desktop IDE.

in such case the PR will be:
- to add `24000` in list static internal ports if `cfg.DesktopIDE != nil`
- move 24000 as an arg to startup code
- probably investigate how it should be integrated in ws-proxy to ensure consistent behaviour with web IDE

Reply:
ok, let's just do changes in supervisor

I tried root path and was getting forbidden, probably IDE forbids it not ws-proxy then